### ISSUE TITLE
Fallback to user primary email when EventPerson/PersonLink email address is deleted

### ISSUE THREAD

**Issue by OmeGak** [indico/indico#2528]
Problem:
`EventPerson`/`PersonLink` data doesn't get updated when linked `User` data do, as it should be since we want to preserve the information within an event regardless what the user do afterwards. This assumption is correct in most cases, but it becomes a problem for the email address.

It's not uncommon that users change their email address for the simple reason they are not used anymore. Yet, the old email address will still show up in all contributions linked to the user and there's nothing they can do but going event by event modifying it manually.

Solution:
If a `EventPerson`/`PersonLink` object is linked to a `User`, instead of storing the email as text, it should reference one of the user's emails. When at some point the referenced email gets removed `PersonLink.email` should resolve to the primary email of the user.

**Comment by ThiefMaster** [comment]
I don't think referencing a specific email using anything but the email itself is a good idea.

But of course we could update the email of associated EventPersons when an email is removed from a User. Not sure if we want this though. Sure, the email in the **old** contribution might not be valid anymore, but it was the email of the speaker at this time. I guess it's a case of preserving historical data vs keeping valid emails/links.

**Comment by OmeGak** [comment]
What's the point of historical email address? What happens in the end is that people check a contribution and want to contact the author, reaching them on the displayed email address, which doesn't exist anymore.

**Comment by ThiefMaster** [comment]
Maybe we could prompt people whether to update their email in contributions when they remove one from their account? I don't think people expect removing an email from their account to update everything in an irreversible way.

**Comment by OmeGak** [comment]
For many users that means updating manually hundreds of contributions in who knows which events throughout several categories. That's not an acceptable solution I would say.

**Comment by ThiefMaster** [comment]
I don't mean per contribution, just a "You are using this email address in N events, would you like to replace it with another one" prompt

**Comment by OmeGak** [comment]
Also, it wouldn't be irreversible with the implementation I proposed since it would be simply a fallback if the linked email is not found for the user anymore.

**Comment by OmeGak** [comment]
That prompt will probably change stuff in management area of events you shouldn't be modifying so I would rather go for a more holistic approach.

**Comment by ThiefMaster** [comment]
> Also, it wouldn't be irreversible with the implementation I proposed since it would be simply a fallback if the linked email is not found for the user anymore. 

Yes, so if I e.g. removed my CERN email from my account for whatever reason it would update all my CERN contributions to show my personal email instead and even when adding the email back I could not undo that.

**Comment by OmeGak** [comment]
That's why I propose fallback instead of update on remove. You add it back, the email that was originally used is displayed again.
